### PR TITLE
Support for headless services

### PR DESCRIPTION
**Pull request by grs**
Fixes issue #51

### CODE REVIEW

File: cmd/service-controller/bridges.go [hunk 1]
```diff
@@ -168,6 +176,19 @@ func (c *Controller) updateServiceBindings(required types.ServiceInterface, port
...
+			if bindings.headless == nil {
+				bindings.headless = required.Headless
+			} else if bindings.headless.Name != required.Headless.Name {
+				bindings.headless.Name = required.Headless.Name
+			} else if bindings.headless.Size != required.Headless.Size {
+				bindings.headless.Size = required.Headless.Size
+			} else if bindings.headless.TargetPort != required.Headless.TargetPort {
+				bindings.headless.TargetPort = required.Headless.TargetPort
+			}
+		} else if bindings.headless != nil {
```
Comment (nicob87) [inline cmd/service-controller/bridges.go:189]: If I am not missing something, you can remove one "if" and just: 
```
                else {
			bindings.headless = nil
		}
```

File: api/types/types.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ var ControllerEditPolicyRule = []rbacv1.PolicyRule{
 	{
 		Verbs:     []string{"get", "list", "watch", "create", "update", "delete"},
 		APIGroups: []string{""},
-		Resources: []string{"services", "configmaps", "pods"},
+		Resources: []string{"services", "configmaps", "pods", "endpoints"},
```
[the remark this answers is not in the document]
Comment (grs) [reply api/types/types.go:85]: Actually I had to abandon that approach and forgot to remove the added permission. I'll remove that.

File: client/van_serviceinterface_update.go [hunk 3]
```diff
@@ -244,6 +244,12 @@ func (cli *VanClient) GetHeadlessServiceConfiguration(targetName string, protoco
 					Name: statefulset.ObjectMeta.Name,
 					Size: int(*statefulset.Spec.Replicas),
 				},
+				Targets: []types.ServiceInterfaceTarget {
```
[the remark this answers is not in the document]
Comment (grs) [reply client/van_serviceinterface_update.go:247]: I'll fix that in this PR though that path is not actually used by headless services, which is why I missed it. That typo was introduced by me in an older commit. Good catch!

**Comment by grs** [comment]
@ajssmith thanks for review! I have fixed those issues now and retested stateful expose with *and without* --headless option.

**Pull request merged**